src/main/java/io/cryostat/net/web/http/api/v1/RecordingDeleteHandler.java

```diff
@@ -102,34 +94,24 @@ public boolean isAsync() {
     @Override
     public void handleAuthenticated(RoutingContext ctx) throws Exception {
         String recordingName = ctx.pathParam("recordingName");
-        fs.listDirectoryChildren(savedRecordingsPath).stream()
-                .filter(saved -> saved.equals(recordingName))
-                .map(savedRecordingsPath::resolve)
-                .findFirst()
-                .ifPresentOrElse(
-                        path -> {
-                            try {
-                                if (!fs.exists(path)) {
-                                    throw new HttpStatusException(404, recordingName);
-                                }
-                                fs.deleteIfExists(path);
-                                notificationFactory
-                                        .createBuilder()
-                                        .metaCategory(NOTIFICATION_CATEGORY)
-                                        .metaType(HttpMimeType.JSON)
-                                        .message(Map.of("recording", recordingName))
-                                        .build()
-                                        .send();
-                            } catch (IOException e) {
-                                throw new HttpStatusException(500, e.getMessage(), e);
-                            } finally {
-                                reportService.delete(recordingName);
-                            }
-                            ctx.response().setStatusCode(200);
-                            ctx.response().end();
-                        },
-                        () -> {
-                            throw new HttpStatusException(404, recordingName);
-                        });
+        try {
+            recordingArchiveHelper.deleteRecording(recordingName).get();
+            notificationFactory
+                    .createBuilder()
+                    .metaCategory(NOTIFICATION_CATEGORY)
+                    .metaType(HttpMimeType.JSON)
+                    .message(Map.of("recording", recordingName))
+                    .build()
+                    .send();
+            ctx.response().setStatusCode(200);
```

Comment:
Just a style thing, but `200` is already the default status code assumed by the `IntermediateResponse` if you don't set one, so explicitly setting it to `200` isn't required but also isn't harmful.

Reply:
I'll remove it then, looks cleaner that way.